### ISSUE TITLE
yum/dnf modules query repos even for removals - can't remove package if repo is broken

### ISSUE THREAD

**Issue by lukurde** [ansible/ansible#81068]
### Summary

Hi,

Running `yum remove foo` works even if repositories are broken.

Running the following task fails:
```
      - name: "Remove the {{ foo }} package"
        ansible.builtin.package:
          name: "{{ foo }}"
          state: absent
```

```
FAILED! => {"changed": false, "msg": "Failed to download metadata for repo '<redacted>': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried", "rc": 1, "results": []}
```

Yes, I know repos should work in general but the `package` module behavior is:
a) inconsistent with the provider (yum in this case)
and
b) inefficient as removals will take unnecessary time to refresh repos

### Issue Type

Bug Report

### Component Name

ansible.builtin.package

### Ansible Version

```console
$ ansible --version
ansible [core 2.15.0]
  config file = /<redacted>/ansible.cfg
  configured module search path = ['/home/<redacted>/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/<redacted>/venv/ansible/lib64/python3.9/site-packages/ansible
  ansible collection location = /home/<redacted>/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/<redacted>/venv/ansible/bin/ansible
  python version = 3.9.13 (main, Nov 16 2022, 15:31:39) [GCC 8.5.0 20210514 (Red Hat 8.5.0-15)] (/home/<redacted>/venv/ansible/bin/python3.9)
  jinja version = 3.1.2
  libyaml = True
```

### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all

CONFIG_FILE() = /home/<redacted>/ansible.cfg
DEFAULT_FORKS(/home/<redacted>/ansible.cfg) = 50
DEFAULT_GATHER_TIMEOUT(/home/<redacted>/ansible.cfg) = 30
DEFAULT_TIMEOUT(/home/<redacted>/ansible.cfg) = 10
DEFAULT_VAULT_PASSWORD_FILE(/home/<redacted>/ansible.cfg) = /home/<redacted>/.vault>
HOST_KEY_CHECKING(/home/<redacted>/ansible.cfg) = False

CONNECTION:
==========

paramiko_ssh:
____________
host_key_checking(/home/<redacted>/ansible.cfg) = False
timeout(/home/<redacted>/ansible.cfg) = 10

ssh:
___
host_key_checking(/home/<redacted>/ansible.cfg) = False
timeout(/home/<redacted>/ansible.cfg) = 10
```

### OS / Environment

Rocky 8.7

### Steps to Reproduce

```
      - name: "Remove the {{ foo }} package"
        ansible.builtin.package:
          name: "{{ foo }}"
          state: absent
```

### Expected Results

`package` module should follow yum - should remove package without querying repositories. Package removal is an offline operation.

### Actual Results

```console
FAILED! => {"changed": false, "msg": "Failed to download metadata for repo '<redacted>': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried", "rc": 1, "results": []}
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/package.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/package.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by lukurde** [comment]
I've reported an issue with the `package` module, not `yum` module. The generic `package` (module) does not have `cacheonly` option. I will have to write tasks using specific modules and conditionals for every distribution family just because **it's impossible to *remove* a package without querying online repositories**. Which is doable but it's not as succinct and clean.

The problem reported is about how Ansible *modules* do not follow the default behavior of *package managers* they're supposed to support. By default `yum` (package manager) does not query repositories and can remove packages completely offline.

I've looked into `apt` module as well, looks like it has completely different default behavior. This is the result of running exact same task using `package` module with `state: absent` on RHEL and Ubuntu (both with disabled repos and cache cleaned):

```
PLAY RECAP **************************************************************************************
<rhel>   : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
<ubuntu> : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0 
```
RHEL: FAILED
Ubuntu: CHANGED

Exact same task, using `package` module gives two completely different results - it's inconsistent and it doesn't match the package manager's default behavior - both `yum` (package manager) and `apt` (package manager) can remove packages without repos/with broken repos/completely offline.

It doesn't make any sense to turn **removal** a clearly OFFLINE operation into ONLINE operation.

I've run the playbook with `-vvvvv`:

```
The full traceback is:
  File "/tmp/ansible_ansible.legacy.dnf_payload_3nohti3n/ansible_ansible.legacy.dnf_payload.zip/ansible/modules/dnf.py", line 669, in _base
  File "/usr/lib/python3.6/site-packages/dnf/base.py", line 405, in fill_sack
    self._add_repo_to_sack(r)
  File "/usr/lib/python3.6/site-packages/dnf/base.py", line 140, in _add_repo_to_sack
    repo.load()
  File "/usr/lib/python3.6/site-packages/dnf/repo.py", line 580, in load
    raise dnf.exceptions.RepoError(str(e))
fatal: [x.x.x.x]: FAILED! => {
```
Looks like you query repositories even for `state: absent` something that the `yum` (package manager) does not do.

Can we reconsider the default behavior of `yum` (module) and stop querying repos on removals to match `yum` (package manager) default behavior and make it consistent?
